server/datastore/mysql/software.go

```diff
@@ -415,18 +415,14 @@ func selectSoftwareSQL(opts fleet.SoftwareListOptions) (string, []interface{}, e
 			"s.arch",
 			"scv.cpe_id", // for join on sub query
 			goqu.COALESCE(goqu.I("scp.cpe"), "").As("generated_cpe"),
+		).
+		Join( // filter software that is not associated with any hosts
+			goqu.I("host_software").As("hs"),
+			goqu.On(
+				goqu.I("hs.software_id").Eq(goqu.I("s.id")),
+			),
```

Comment:
Something to double-check: `selectSoftwareSQL` uses `software_host_counts` when `opts.WithHostCounts == true`. 

1. Do we even need the `software_host_counts` table now? (Let's check why we add it in the first place.)
2. Does the query result continue make sense with this change?

@chiiph

Reply:
> Do we even need the software_host_counts table now? (Let's check why we add it in the first place.)

We do, we are still counting and showing the counts.

> Does the query result continue make sense with this change?

Good question. I _think_ it does, I wonder how it'll perform, though.